swift-tools-support-core/Sources/TSCTestSupport/PseudoTerminal.swift

```diff
@@ -31,10 +31,14 @@ public final class PseudoTerminal {
     }
     
     public func readMaster(maxChars n: Int = 1000) -> String? {
-        var buf: [CChar] = [CChar](repeating: 0, count: n)
-        if read(master, &buf, n) <= 0 {
-            return nil
+        let buf: [CChar] = .init(unsafeUninitializedCapacity: n + 1 /* +1 for null terminator */ ) {
+            buf, initializedCapacity in
+            initializedCapacity = read(master, &buf, n)
+            guard initializedCapacity > 0 else { return }
```

Comment:
Looks good now, the only remaining question is how happy `unsafeUninitializedCapacity:` is if you return a negative number by reference (as it would be if `read()` returned an error).  So maybe it should be set to zero before returning.  I'm not familiar enough with `unsafeUninitializedCapacity:` to know if it's allowed to be negative on return; the documentation says "Inside the closure, set the initializedCount parameter to the number of elements that are initialized by the closure." and doesn't address a negative count, so I assume it's disallowed.

Apart from that it looks good!

Reply:
Good point! Fixed that as well.